worker/src/RequestBodyBuffer/RequestBodyBuffer_Remote.ts

```diff
@@ -138,9 +138,35 @@ export class RequestBodyBuffer_Remote implements IRequestBodyBuffer {
     );
   }
 
-  public tempSetBody(body: string): void {
+  async bodyLength(): Promise<number> {
+    try {
+      await this.ingestPromise.catch(() => undefined);
+      const response = await this.requestBodyBuffer.fetch(
+        `${BASE_URL}/${this.uniqueId}/body-length`,
+        { method: "GET" }
+      );
+      if (!response.ok) {
+        return 0;
+      }
+      const json = await response.json<{ length: number }>();
+      return json.length;
+    } catch (e) {
+      console.error("RequestBodyBuffer_Remote bodyLength error", e);
+      return 0;
+    }
+  }
+
+  public async tempSetBody(body: string): Promise<void> {
```

Comment:
i wasnt going to do this, cuz i was gunna rewrite all the ai-gateway and prompt stuff but for now let's just do this to be safe